godot_project/editor/input_handlers/molecular_structures/selection_input_handler.gd

```diff
@@ -115,33 +115,62 @@ func forward_input(in_input_event: InputEvent, in_camera: Camera3D, in_context:
 		var hovering_atom_id: int = -1
 		var hovering_bond_id: int = -1
 		var hovering_spring_id: int = -1
+		var hover_position: Vector3 = Vector3(INF, INF, INF)
 		if not editable_structures.is_empty():
 			var multi_structure_hit_result := MultiStructureHitResult.new(in_camera, in_input_event.position, editable_structures)
 			match multi_structure_hit_result.hit_type:
 				MultiStructureHitResult.HitType.HIT_ATOM:
 					hovering_object = multi_structure_hit_result.closest_hit_structure_context
 					hovering_atom_id = multi_structure_hit_result.closest_hit_atom_id
+					hover_position = hovering_object.nano_structure.atom_get_position(hovering_atom_id)
 				MultiStructureHitResult.HitType.HIT_BOND:
 					hovering_object = multi_structure_hit_result.closest_hit_structure_context
 					hovering_bond_id = multi_structure_hit_result.closest_hit_bond_id
+					var bond_data: Vector3i = hovering_object.nano_structure.get_bond(hovering_bond_id)
+					hover_position = (hovering_object.nano_structure.atom_get_position(bond_data.x) + \
+							hovering_object.nano_structure.atom_get_position(bond_data.y)) / 2.0
 				MultiStructureHitResult.HitType.HIT_SPRING:
 					hovering_object = multi_structure_hit_result.closest_hit_structure_context
 					hovering_spring_id = multi_structure_hit_result.closest_hit_spring_id
-				MultiStructureHitResult.HitType.HIT_MOTOR, MultiStructureHitResult.HitType.HIT_ANCHOR:
+					hover_position = (hovering_object.nano_structure.spring_get_atom_position(hovering_spring_id) + \
+							hovering_object.nano_structure.spring_get_anchor_position(hovering_spring_id, hovering_object)) / 2.0
+				MultiStructureHitResult.HitType.HIT_MOTOR:
+					hovering_object = multi_structure_hit_result.closest_hit_structure_context
+					hover_position = hovering_object.nano_structure.get_transform().origin
+				MultiStructureHitResult.HitType.HIT_ANCHOR:
 					hovering_object = multi_structure_hit_result.closest_hit_structure_context
+					hover_position = hovering_object.nano_structure.get_position()
 				MultiStructureHitResult.HitType.HIT_SHAPE:
 					if _is_shape_selectable():
 						hovering_object = multi_structure_hit_result.closest_hit_structure_context
+					hover_position = hovering_object.nano_structure.get_position()
 				_:
 					# do nothing
 					pass
 		get_workspace_context().set_hovered_structure_context(hovering_object, hovering_atom_id, hovering_bond_id, hovering_spring_id)
+		var selection_center: Vector3 = workspace_context.get_selection_aabb().get_center() if \
+				workspace_context.has_selection() else Vector3(INF, INF,INF)
+		_update_distance_message(workspace_context, hover_position, selection_center)
```

Comment:
i dont think it should show the distance to an arbitrary center of selection.
Distance should be contextual, in example
- if oyu select an atom it should show distance to another atom that you hover
- if you select a bond, and hover another bond, and both bonds share an atom, it should show the angle between both

Reply:
I don't have opinion on those, in practice those are probably questions to @jmackley100